src/task/blocking.rs

```diff
@@ -73,25 +100,53 @@ fn maybe_create_another_blocking_thread() {
         .spawn(move || {
             let wait_limit = Duration::from_millis(1000 + rand_sleep_ms);
 
-            DYNAMIC_THREAD_COUNT.fetch_add(1, Ordering::Relaxed);
+            CURRENT_POOL_SIZE.fetch_add(1, Ordering::Relaxed);
             while let Ok(task) = POOL.receiver.recv_timeout(wait_limit) {
                 abort_on_panic(|| task.run());
+                calculate_dispatch_frequency();
             }
-            DYNAMIC_THREAD_COUNT.fetch_sub(1, Ordering::Relaxed);
+            CURRENT_POOL_SIZE.fetch_sub(1, Ordering::Relaxed);
```

Comment:
This can't be a Relaxed atomic if we want to guarantee a happens-before relationship with the addition, which is important to guarantee that we never divide by zero in the ema calculation above

Reply:
(can also just add 1 to the divisor in the calculation above to enforce this invariant at the dependency site)

Reply:
but atomics will wrap, so we probably want to use a proper barrier here anyway for defense against wrap-intolerance in other calculation areas

Reply:
Ordering::Release for writes generally and Ordering::Acquire for all loads I think is OK on this code, but honestly due to the timescales involved in blocking codepaths it's fine to use a SeqCst to think even less about wrap issues.

Reply:
I am going to use SeqCst in both.